pkg/zabbix_api.go

```diff
@@ -112,16 +112,22 @@ func (ds *ZabbixDatasource) BuildResponse(result *simplejson.Json) (*datasource.
 func (ds *ZabbixDatasource) ZabbixRequest(ctx context.Context, dsInfo *datasource.DatasourceInfo, method string, params *simplejson.Json) (*simplejson.Json, error) {
 	zabbixUrl := dsInfo.GetUrl()
 
-	// Authenticate first
-	if zabbixAuth == "" {
+	var result *simplejson.Json
+	var err error
+
+	for attempt := 0; attempt <= 3; attempt++ {
+		// Authenticate
 		auth, err := ds.loginWithDs(ctx, dsInfo)
```

Comment:
In this case, data source tries to login at every request. We should use stored `zabbixAuth` token and renew it only if "not authorized" error happens.

Reply:
Good catch, Updated https://github.com/alexanderzobnin/grafana-zabbix/pull/808/commits/371c68dd2e20fe720f565785000145f09ec6f539